ldes-server-infra-postgres/postgres-ingest-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/postgres/MemberPostgresRepository.java

```diff
@@ -36,15 +36,17 @@ public MemberPostgresRepository(MemberEntityRepository repository,
 	@Transactional
 	public int insertAll(List<IngestedMember> members) {
 		final int collectionId = getCollectionId(members.getFirst().getCollectionName());
+
 		if (membersContainDuplicateIds(members)) {
 			return 0;
 		}
-		String sql = """
-				INSERT INTO members (subject, collection_id, version_of, timestamp, transaction_id, member_model)
-				VALUES (?,?,?,?,?,?)
-				""";
+
 		try {
-			final List<Object[]> batchArgs = members.stream()
+			var toBatchMembers = members.stream()
+					.filter(ingestedMember -> !ingestedMember.equals(members.getLast()))
+					.toList();
+
+			final List<Object[]> batchArgs = toBatchMembers.stream()
```

Comment:
Why is this collected into a list, as in the next step it is used as a stream again